blob/gcsblob/gcsblob.go

```diff
@@ -230,6 +306,56 @@ func OpenBucket(ctx context.Context, client *gcp.HTTPClient, bucketName string,
 	return blob.NewBucket(drv), nil
 }
 
+// credentialsClient is a wrapper for lazy initialization.
+type credentialsClient struct {
+	init   sync.Once
+	client *credentials.IamCredentialsClient
+	err    error
+}
+
+// errPassthrough satisfies the signature expected by storage.SignedURLOptions.SignBytes
+// but really just passes through any permanent error.
+// It's extracted to minimize allocations.
+func (c *credentialsClient) errPassthrough(_ []byte) ([]byte, error) {
+	return nil, c.err
+}
+
+// CreateMakeSignBytesWith produces a MakeSignBytes variant from an expanded parameter set.
+// It essentially adapts a remote call to the IAM Credentials API
+// to the function signature expected by storage.SignedURLOptions.SignBytes.
+func (c *credentialsClient) CreateMakeSignBytesWith(lifetimeCtx context.Context, googleAccessID string) func(context.Context) SignBytesFunc {
+	return func(requestCtx context.Context) SignBytesFunc {
+		c.init.Do(func() {
+			c.client, c.err = credentials.NewIamCredentialsClient(lifetimeCtx)
+		})
+
+		if c.err != nil {
+			return c.errPassthrough
+		}
+
+		return func(p []byte) ([]byte, error) {
+			resp, err := c.client.SignBlob(
+				requestCtx,
+				&credentialspb.SignBlobRequest{
+					Name:    googleAccessID,
+					Payload: p,
+				})
+			if err != nil {
+				return nil, err
+			}
+			return resp.GetSignedBlob(), nil
+		}
+	}
+}
+
+// Close will sever any established connection to the IAM Credentials API.
+// Call this when the corresponding bucket gets closed.
```

Comment:
Comment about calling it when the bucket gets closed is stale; this credentialsClient is created once per application for the initOnce of the default URLOpener.

Reply:
(Can probably just delete this function entirely)